spring-cloud-stream-metrics/src/main/java/org/springframework/cloud/stream/metrics/config/MetricJsonSerializer.java

```diff
@@ -40,7 +40,12 @@
 @JsonComponent
 public class MetricJsonSerializer {
 
-	final static DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+	final static ThreadLocal<DateFormat> df = new ThreadLocal<DateFormat>(){
+			@Override
+			protected DateFormat initialValue() {
+					return new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+			}
+	};
```

Comment:
I'll make the changes, we may need to change the formatter itself, on the collector we changed it to round values to 2 decimal places. After this milestone of the collector, I want to make sure it uses the code that lives on stream-metrics, instead of duplicating it, too much duplicate effort. It was good to allow us to walk at a different pace, but after M1 I want to revisit this design.